File: 268. Missing Number/Solution.java
class Solution {
    public int missingNumber(int[] nums) {
    	if(nums == null || nums.length == 0){
    		return 0;
    	}
    	int missingNum = nums.length;
    	for(int i = 0;i < nums.length;i++){
    		missingNum ^= nums[i] ^ i;
    	}
    	return missingNum;
    }
}

File: 268. Missing Number/Solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    	if(!nums){
			return 0;
		}
    const res = new Array(nums.length + 1).fill(-1);
    for(const num of nums){
        res[num] = num;
    }
    return res.indexOf(-1);
};
